src/frames.ts

```diff
@@ -693,16 +693,34 @@ export class Frame {
     return result!;
   }
 
+  private async _retryWithSelectorIfNotConnected<R>(
+    selector: string, options: types.PointerActionWaitOptions,
+    action: (handle: dom.ElementHandle<Element>, deadline: number) => Promise<R>): Promise<R> {
+    const deadline = this._page._timeoutSettings.computeDeadline(options);
+    while (!helper.isPastDeadline(deadline)) {
+      try {
+        const { world, task } = selectors._waitForSelectorTask(selector, 'attached', deadline);
+        const handle = await this._scheduleRerunnableTask(task, world, deadline, `selector "${selector}"`);
+        const element = handle.asElement() as dom.ElementHandle<Element>;
+        const result = await action(element, deadline);
+        element.dispose();
+        return result;
+      } catch (e) {
+        if (options.force || !(e instanceof NotConnectedError))
+          throw e;
```

Comment:
it'd be nice to dispose the handle to the detached element - we are likely the only client retaining it.

Reply:
Done.